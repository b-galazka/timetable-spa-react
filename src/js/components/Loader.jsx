import React from 'react';
import { connect } from 'react-redux';

import LoadingAnimation from './LoadingAnimation';
import errors from '../../json/errors';

import '../../scss/noContent.scss';

function mapStateToProps(state) {

    const { fetchingError } = state.initialState;

    return {
        initialStateFetchingError: fetchingError
    };
}

function Loader({ initialStateFetchingError }) {

    return (
        <section className="app__no-content">

            {
                (initialStateFetchingError) ?

                <p className="app__fetching-error">{errors.fetchingError}</p> :

                <figure>
                    <LoadingAnimation width="120px" height="120px" />
                </figure>
            }

        </section>
    );
}

export default connect(mapStateToProps)(Loader);