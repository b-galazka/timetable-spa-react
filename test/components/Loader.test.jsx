import React from 'react';
import renderer from 'react-test-renderer';

import Loader from '../../src/js/components/Loader';

jest.mock('../../src/js/components/LoadingAnimation', () => 'LoadingAnimation');
jest.mock('../../src/json/errors', () => ({ fetchingError: 'fetching error' }));

describe('Loader component', () => {

    it('should render loading animation if there is no fetching error', () => {

        const tree = renderer.create(
            <Loader.WrappedComponent
                initialStateFetchingError={false}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render fetching error', () => {

        const tree = renderer.create(
            <Loader.WrappedComponent
                initialStateFetchingError
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});