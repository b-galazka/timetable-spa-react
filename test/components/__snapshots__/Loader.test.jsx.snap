// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render fetching error 1`] = `
<section
  className="app__no-content"
>
  <p
    className="app__fetching-error"
  >
    fetching error
  </p>
</section>
`;

exports[`Loader component should render loading animation if there is no fetching error 1`] = `
<section
  className="app__no-content"
>
  <figure>
    <LoadingAnimation
      height="120px"
      width="120px"
    />
  </figure>
</section>
`;
